src/components/molecules/Visualizer/Plugin/types.ts

```diff
@@ -189,9 +190,17 @@ export type Visualizer = {
   readonly property?: any;
 };
 
+type Rect = {
+  north: number;
+  south: number;
+  east: number;
+  west: number;
+};
+
 export type Camera = {
   /** Current camera position */
   readonly position: CameraPosition | undefined;
+  readonly viewport: () => Rect | undefined;
```

Comment:
The plugin API often uses getters, so make sure to return the value directly instead of returning a function.

```suggestion
  readonly viewport: Rect | undefined;
```